contracts/DisputableDelay.sol

```diff
@@ -0,0 +1,264 @@
+        delayedScript.pausedAt = getTimestamp64();
+        delayedScript.delayedScriptStatus = DelayedScriptStatus.Paused;
+
+        emit ExecutionPaused(_delayedScriptId, delayedScript.actionId);
+    }
+
+    /**
+    * @dev Allow script execution, IDisputable interface conformance
+    * @param _delayedScriptId The ID of the script execution to allow
+    */
+    function _onDisputableActionAllowed(uint256 _delayedScriptId) internal {
+        DelayedScript storage delayedScript = delayedScripts[_delayedScriptId];
+        uint64 timePaused = getTimestamp64().sub(delayedScript.pausedAt);
+
+        delayedScript.executionFromTime = delayedScript.executionFromTime.add(timePaused);
+        delayedScript.pausedAt = 0;
+        delayedScript.delayedScriptStatus = DelayedScriptStatus.Active;
+
+        emit ExecutionResumed(_delayedScriptId, delayedScript.actionId);
+    }
+
+    /**
+    * @dev Reject script execution, IDisputable interface conformance
+    * @param _delayedScriptId The ID of the script execution to reject
+    */
+    function _onDisputableActionRejected(uint256 _delayedScriptId) internal {
+        DelayedScript storage delayedScript = delayedScripts[_delayedScriptId];
+
+        delayedScript.delayedScriptStatus = DelayedScriptStatus.Cancelled;
+
+        emit ExecutionCancelled(_delayedScriptId, delayedScript.actionId);
+    }
+
+    /**
+    * @dev Void script execution, IDisputable interface conformance
+    * @param _delayedScriptId The ID of the script execution to void
+    */
+    function _onDisputableActionVoided(uint256 _delayedScriptId) internal {
+        _onDisputableActionAllowed(_delayedScriptId);
+    }
+
+    /**
+    * @notice Execute the script with ID `_delayedScriptId`
+    * @param _delayedScriptId The ID of the script to execute
+    */
+    function execute(uint256 _delayedScriptId) external nonReentrant {
+        DelayedScript storage delayedScript = delayedScripts[_delayedScriptId];
+        require(_canExecute(delayedScript), ERROR_CANNOT_EXECUTE);
+
+        (,,,,,bool closed,,) = _ensureAgreement().getAction(delayedScript.actionId);
+        if (!closed) {
+            _closeAgreementAction(delayedScript.actionId);
+        }
+
+        delayedScript.delayedScriptStatus = DelayedScriptStatus.Executed;
+
+        address[] memory blacklist = new address[](1);
+        blacklist[0] = address(_getAgreement());
+        runScript(delayedScript.evmCallScript, new bytes(0), blacklist);
+
+        emit ExecutedScript(_delayedScriptId, delayedScript.actionId);
+    }
+
+    /**
+    * @notice Close agreement action with ID `_delayedScriptId`
+    * @param _delayedScriptId The ID of the script to close the agreement action of
+    */
+    function closeAgreementAction(uint256 _delayedScriptId) external {
+        DelayedScript storage delayedScript = delayedScripts[_delayedScriptId];
+
+        require(_canClose(delayedScript), ERROR_CANNOT_CLOSE);
```

Comment:
This one is already checked by the Agreement app on `Agreement#closeAction`

Reply:
You suggested adding _canClose() here: https://github.com/1Hive/delay-app/pull/60#discussion_r439704722

Reply:
I know but we've changed it again, please don't hate us! 🙈 
Now basically `closeAction` and `challengeAction` entry points of the Agreement app rely on their respective checks: `canClose` and `canChallenge`

Reply:
No worries. I think I will remove this function anyway, provided you can close the action via Agreements if the action can't be executed.